### PR TITLE
Epic/vlan pool/tests

### LINKED ISSUE #108: Create a new test to validate the usage and persistency of interface available_tags
Is the available_tags persistent among Kytos resets?

Is mef_eline following the restriction on an interface available tags?

What happens if there is no available tag?

### LINKED ISSUE #257: Modify and create tests related to VLAN pool
VLAN pool is going to change and it requires modification and the addition of tests.

### PR DESCRIPTION
**Pull request by Alopalao**
Closes #257 
Closes #108 

### Summary

Added tests for `vlan_pool` epic, what is being tested:
- EVC allocating tags to interfaces
- Non-duplicated UNI tags when creating and patching
- Setting `tag_ranges` to interfaces including and not (expected error) all the used tags
- Resizing `available_tags` when setting new `tag_ranges`

This PR relies in non-master branches:
- branch_mef_eline=epic/vlan_pool
- branch_of_lldp=epic/vlan_pool
- branch_topology=epic/vlan_pool_request_get
- branch_kytos=epic/vlan_pool

### End-To-End Tests
```
+ python3 -m pytest tests/test_e2e_15_mef_eline.py --reruns 0 -r fEr
============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-7.2.0, pluggy-1.3.0
rootdir: /tests
plugins: timeout-2.1.0, rerunfailures-10.2, anyio-3.6.2
collected 4 items

tests/test_e2e_15_mef_eline.py ....                                      [100%]
```

### CODE REVIEW

File: tests/test_e2e_15_mef_eline.py [hunk 1]
```diff
@@ -221,3 +221,158 @@ def test_001_create_update_with_constraints(self):
...
+        api_url = KYTOS_API + '/mef_eline/v2/evc/'
+        response = requests.post(api_url, json=evc_1)
+        assert response.status_code == 201, response.text
+        assert 'circuit_id' in response.json()
+
+        # Verify if EVC tag has been allocated
+        topo_url = KYTOS_API + "/topology/v3/interfaces/tag_ranges"
+        response = requests.get(topo_url)
+        data = response.json()
+        actual = data["00:00:00:00:00:00:00:02"]["1"]["available_tags"]["vlan"]
```
Comment (viniarck) [inline tests/test_e2e_15_mef_eline.py:248]: - If you could also assert `"00:00:00:00:00:00:00:02:2"` `available_tags` that'd be great. 
- Also if you could take the opportunity to also assert the default `tag_ranges` in the response here it'd be great too

btw I've just asked a minor change on `topology`'s PR https://github.com/kytos-ng/topology/pull/166/files#r1344458662

File: tests/test_e2e_15_mef_eline.py [hunk 2]
```diff
@@ -221,3 +221,158 @@ def test_001_create_update_with_constraints(self):
...
+                "tag": {"tag_type": 1, "value": 200},
+                "interface_id": "00:00:00:00:00:00:00:01:1",
+            },
+            "uni_z": {
+                "tag": {"tag_type": 1, "value": 200},
+                "interface_id": "00:00:00:00:00:00:00:02:2",
+            }
+        }
+        response = requests.post(api_url, json=evc_2)
+        assert response.status_code == 201, response.text
```
Comment (viniarck) [inline tests/test_e2e_15_mef_eline.py:266]: After this point it'd be great to also assert both interfaces `available_tags` response again confirming the new allocations.

File: tests/test_e2e_15_mef_eline.py [hunk 3]
```diff
@@ -221,3 +221,158 @@ def test_001_create_update_with_constraints(self):
...
+
+        # Patch EVC with used tag value
+        payload = {
+            "uni_a": {
+                "tag": {"tag_type": 1, "value": 100},
+                "interface_id": "00:00:00:00:00:00:00:02:1",
+            },
+        }
+        response = requests.patch(api_url+evc_2_id, json=payload)
+        assert response.status_code == 400, response.text
```
Comment (viniarck) [inline tests/test_e2e_15_mef_eline.py:279]: After this point for completeness if you could also assert that `available_tags` didn't change that'd be great.

**Review by viniarck — COMMENTED** [review]
Excellent initial coverage @Alopalao, fantastic show this feature turned out. 

I asked a few extra asserts a minor API response change though. Thanks.

File: tests/test_e2e_15_mef_eline.py (hunk 1 above)
Comment (Alopalao) [reply tests/test_e2e_15_mef_eline.py]: Updated this part due to changes in `topology`'s PR.
Added confirmation of changes for `"00:00:00:00:00:00:00:02:2"` in `available_tags` and `tag_ranges` as well as in the other used interfaces.
Added used branches in the top comment.

File: tests/test_e2e_15_mef_eline.py (hunk 2 above)
Comment (Alopalao) [reply tests/test_e2e_15_mef_eline.py:273]: Added assertion.

File: tests/test_e2e_15_mef_eline.py (hunk 3 above)
Comment (Alopalao) [reply tests/test_e2e_15_mef_eline.py:309]: Added verification.

**Review by viniarck — APPROVED** [review]
Leaving it pre-approved @Alopalao. 

Excellent how the new endpoints and coveraged turned out. Let me also explicitly include @italovalcy as a reviewer since he'll probably like to review this one too.

**Pull request merged**